### PR TITLE
patrons: add a role async validator

### PR DESCRIPTION
**Pull request by jma**
* Removes useless console log messages.
* Adds a role patron editor validator to avoid multiple professional
  accounts

## For reviewer

Only the last commit.

## Why are you opening this PR?

- https://tree.taiga.io/project/rero21-reroils/us/2034?milestone=289558

## Dependencies

- https://github.com/rero/rero-ils/pull/1797

## How to test?

- What command should I have to run to test your PR?
- What should I test through the UI?

## Code review check list

- [ ] Commit message template compliance.
- [ ] Commit message without typos.
- [ ] File names.
- [ ] Functions names.
- [ ] Functions docstrings.
- [ ] Unnecessary commited files?
- [ ] Cypress tests successful?

### CODE REVIEW

File: projects/admin/src/app/routes/patrons-route.ts [hunk 1]
```diff
@@ -103,6 +111,44 @@ export class PatronsRoute extends BaseRoute implements RouteInterface {
       }
     };
   }
+  /**
+   * Create an Async validator to avoid multiple organisation prfessional accouts.
+   *
+   * @param field - formly field config.
+   */
+   _getRoleValidator(field: FormlyFieldConfig) {
```
Comment (Garfield-fr) [inline projects/admin/src/app/routes/patrons-route.ts:119]: add term `private`

File: projects/admin/src/app/routes/patrons-route.ts [hunk 2]
```diff
@@ -103,6 +111,44 @@ export class PatronsRoute extends BaseRoute implements RouteInterface {
...
+    return {
+      expression: (control: FormControl) => {
+        console.log('validate');
```
Comment (Garfield-fr) [inline projects/admin/src/app/routes/patrons-route.ts:122]: remove this

**Review by Garfield-fr — APPROVED** [review]

File: projects/admin/src/app/routes/patrons-route.ts [hunk 3]
```diff
@@ -103,6 +111,43 @@ export class PatronsRoute extends BaseRoute implements RouteInterface {
       }
     };
   }
+  /**
+   * Create an Async validator to avoid multiple organisation prfessional accouts.
+   *
+   * @param field - formly field config.
+   */
+   private _getRoleValidator(field: FormlyFieldConfig) {
```
Comment (Garfield-fr) [inline projects/admin/src/app/routes/patrons-route.ts:119]: output typing: `Observable<Boolean>` ?

**Review by Garfield-fr — APPROVED** [review]

File: projects/admin/src/app/routes/patrons-route.ts [hunk 4]
```diff
@@ -103,6 +111,43 @@ export class PatronsRoute extends BaseRoute implements RouteInterface {
...
+            const patronAccounts = user.metadata.patrons;
+            if (patronAccounts && patronAccounts.length > 0) {
+              // accounts in other organisations
+              return !patronAccounts.some(ptrn => {
+                if (ptrn.organisation.pid !== currentOrgPid) {
+                  if (ptrn.roles.some(role => ['librarian', 'system_librarian'].some(r => r === role))) {
+                    return true;
+                  }
+                }
+                return false;
```
Comment (zannkukai) [inline projects/admin/src/app/routes/patrons-route.ts:142]: merged ? 
```javascript
return (ptrn.organisation.pid !== currentOrgPid) && (ptrn.roles.some(role => ['librarian', 'system_librarian'].some(r => r === role)));
```

**Review by zannkukai — APPROVED** [review]

**Pull request merged**